DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -861,15 +954,64 @@ main()
 	
 	# reapply multirun fix
 	
-	for (( i = 0; i < ${#mrFIXConcatNames[@]}; ++i ))
-	do
-	    log_Msg "ReApply MultiRun FIX Cleanup"
-	    log_Msg "mrFIXNames: ${mrFIXNames[$i]}"
-	    log_Msg "mrFIXConcatNames: ${mrFIXConcatNames[$i]}"
-	    #stage 2 parsing is done by reapply script
-	    reapply_mr_fix_cmd=("${HCPPIPEDIR}/ICAFIX/ReApplyFixMultiRunPipeline.sh" --path="${StudyFolder}" --subject="${Subject}" --fmri-names="${mrFIXNames[$i]}" --concat-fmri-name="${mrFIXConcatNames[$i]}" --high-pass="${HighPass}" --reg-name="${ConcatRegName}" --matlab-run-mode="${MatlabRunMode}" --motion-regression="${MotionRegression}")
-	    log_Msg "reapply_mr_fix_cmd: ${reapply_mr_fix_cmd[*]}"
-	    "${reapply_mr_fix_cmd[@]}"
+    for (( i = 0; i < ${#mrFIXConcatNames[@]}; ++i ))
+    do
+        log_Msg "ReApply MultiRun FIX Cleanup"
+        log_Msg "mrFIXNames: ${mrFIXNames[$i]}"
+        log_Msg "mrFIXConcatNames: ${mrFIXConcatNames[$i]}"
+        #stage 2 parsing is done by reapply script
+        reapply_mr_fix_cmd=("${HCPPIPEDIR}/ICAFIX/ReApplyFixMultiRunPipeline.sh" --path="${StudyFolder}" --subject="${Subject}" --fmri-names="${mrFIXNames[$i]}" --concat-fmri-name="${mrFIXConcatNames[$i]}" --high-pass="${HighPass}" --reg-name="${ConcatRegName}" --matlab-run-mode="${MatlabRunMode}" --motion-regression="${MotionRegression}")
+        log_Msg "reapply_mr_fix_cmd: ${reapply_mr_fix_cmd[*]}"
+        "${reapply_mr_fix_cmd[@]}"
+        
+        for regname in "$ConcatRegName" "${extractExtraRegNamesArr[@]+"${extractExtraRegNamesArr[@]}"}"
+        do
+            #MSMSulc special naming convention
+            if [[ "$regname" == "MSMSulc" ]]
+            then
+                regname=""
+                regstring=""
+            else
+                regstring=_"$regname"
+            fi
+            
+            extract_cmd=("${HCPPIPEDIR}/global/scripts/ExtractFromMRFIXConcat.sh"
+                            --study-folder="$StudyFolder"
+                            --subject="$Subject"
+                            --multirun-fix-names="${mrFIXNames[$i]}"
+                            --csv-out="$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXConcatNames[$i]}/${mrFIXConcatNames[$i]}_Runs.csv"
+                            --concat-cifti-input="$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXConcatNames[$i]}/${mrFIXConcatNames[$i]}_Atlas${regstring}_hp${HighPass}_clean.dtseries.nii"
+                            --surf-reg-name="$regname")
+            
+            if (( ${#mrFIXExtractConcatNamesArr[@]} > 0 )) && [[ "${mrFIXExtractConcatNamesArr[$i]}" != NONE ]]
+            then
+                mkdir -p "$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXExtractConcatNamesArr[$i]}"
+                
```

Comment:
Still think we should have a comment here:
```suggestion
                # Using clean_vn.dscalar from the full concat group for the extracted concat group as well.
                # (i.e., estimate for the variance normalization map is based on the full concat group, not 
                # the subset of extracted scans)
```